dotnet/src/Azure.Iot.Operations.Protocol/RPC/CommandInvoker.cs

```diff
@@ -575,6 +576,25 @@ public async Task<ExtendedResponse<TResp>> InvokeCommandAsync(TReq request, Comm
                     requestMessage.ContentType = payloadContext.ContentType;
                 }
 
+                if (metadata?.CloudEvent is not null)
+                {
+                    // if type has default value then user did not set it, so set it to rpc.request
+                    if (metadata.CloudEvent.Type == "ms.aio.telemetry")
+                    {
+                        var newCloudEvent = new CloudEvent(metadata.CloudEvent.Source, "ms.aio.rpc.request", metadata.CloudEvent.SpecVersion)
```

Comment:
Cloning like this usually results in bugs later if we add more fields to the CloudEvent type (because we likely forget to update lines 586 to 589 to include copying over the new fields as well).

In cases like this, it is fine to just alter the object that the user passed in like

```csharp
metadata.CloudEvent.Type ??= "ms.aio.rpc.request"
```

Same for assigning the default values in the telemetry sender + command executor